src/Message/ItemDataTrait.php

```diff
@@ -19,18 +19,25 @@ public function getItemData(ItemBag $items): array
 
         foreach ($items as $item) {
             $taxRate = $item->getTaxRate();
+            $price = null === $item->getPrice() ? $this->convertToMoney(0) : $this->convertToMoney($item->getPrice());
             $totalTaxAmount = null === $item->getTotalTaxAmount()
                 ? $this->convertToMoney(0)
                 : $this->convertToMoney($item->getTotalTaxAmount());
-            $price = null === $item->getPrice() ? $this->convertToMoney(0) : $this->convertToMoney($item->getPrice());
+            $totalDiscountAmount = null === $item->getTotalDiscountAmount()
+                ? $this->convertToMoney(0)
+                : $this->convertToMoney($item->getTotalDiscountAmount());
+            $totalAmount = null === $item->getTotalAmount()
```

Comment:
Klarna specifies this must match `(quantity * unit_price) - total_discount_amount within ±quantity`, which is no longer guaranteed. Would the following also cover your use-case?
```
'total_amount' => (int) $price->multiply($item->getQuantity())->subtract($totalDiscountAmount)->getAmount(),
```

Reply:
Yes, I ran my own codebase unit tests and Klarna accepts the default (now `$price->multiply($item->getQuantity())->subtract($totalDiscountAmount)`) when I don't manually override/supply a `total_amount` for the item. 👍